### PR TITLE
Create UI for adding and editing Collimator component in Instrument Editor

### PR DESCRIPTION
**Pull request by DrPaulSharp**
* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce(Bug fix, feature, docs update, ...)?** 
Adds UI for creating/editing collimators in the Instrument Editor

* **What is the current behaviour (You can also link to an open issue here)?**
Collimators can only be edited by editing the JSON directly in the Instrument Editor. There are UI components for only the general, instrument jaws and detectors.

* **What is the new behaviour (if this is a feature change)?**
Collimators can be edited using the UI component as well as editing the JSON

* **Does this PR introduce a breaking change (What changes might users need to make in their application due to this PR)?**
No

* **Other information:**
None

### CODE REVIEW

File: sscanss/editor/designer.py [hunk 1]
```diff
@@ -955,3 +959,256 @@ def value(self):
...
+            detector_name = data.get('name', '')
+            if detector_name:
+                detectors.append(detector_name)
+        self.detector_combobox.clear()
+        self.detector_combobox.addItems([*detectors])
+
+        # Choose a detector if at least one is listed, otherwise display a warning
+        if detectors:
+            detector = collimator_data.get('detector')
+            if isinstance(detector, str):
```
Comment (StephenNneji) [inline sscanss/editor/designer.py:1166]: the `get` method returns `None` by default. I think its more readable to test detector is not `None`

File: sscanss/editor/designer.py [hunk 2]
```diff
@@ -955,3 +959,256 @@ def value(self):
...
+                self.detector_combobox.setCurrentText(detector)
+            else:
+                self.detector_combobox.setCurrentIndex(0)
```
Comment (StephenNneji) [inline sscanss/editor/designer.py:1169]: If the user types in a new detector, the first detector is always selected which is confusing. It might be better to make the combobox editable so the user can type in any detector name (although they would have to go back and create the detector). Also with the editable combobox you don't need the **no detector** warning but will still have to check that the detector value is not empty. What do you think?

Comment (DrPaulSharp) [reply sscanss/editor/designer.py:1169]: Yeah, this all sounds good. The big hole is of course that a detector can be specified without existing in the file, but there is a warning given about that, and this way of doing things has the advantage that the detector and collimators can be added in either order.

Comment (DrPaulSharp) [reply sscanss/editor/designer.py:1169]: One thing to consider is in a new detector is added via a collimator, but is not defined within the detectors. In this case should the new detector be available as an option within the combobox in the collimator component, or should the combobox be restricted to detectors defined in the detector part of the JSON?
